client/src/services/apiProxy.js

```diff
@@ -119,7 +120,7 @@ export async function streamResult(url, name, cb, errCb) {
             const item = await request(`${url}/${name}`);
 
             if (isCancelled) return;
-            cb(item);
+            _.debounce((item)=>{cb(item)}, 250, {leading: true})(item);
```

Comment:
Thanks again for all the work on this! This is really close now, but I think we need one more small change. `_.debounce` will actually create a new function and we'll want to call that same function multiple times. So, I think we need something like following around line number 121

```
const debouncedCallback = _.debounce(item, 250, {leading: true});
```

Then line number 122 becomes 
```
debouncedCallback(item);
````

Also, line number 128 will need to become
```
socket = stream(watchUrl, x => debouncedCallback(x.object), {isJson: true});
```

This debounce function is really confusing. I only know because I had to struggle through getting it right in another project recently ;). Let me know if that doesn't make sense and I'm happy to discuss more in-depth.

Reply:
Hi Eric, I pushed a fix according to CRs.